cgrok/grokre.c

```diff
@@ -14,8 +14,8 @@
 
 /* global, static variables */
 
-#define CAPTURE_ID_LEN 4
-#define CAPTURE_FORMAT "%04x"
+#define CAPTURE_ID_LEN 5
+#define CAPTURE_FORMAT "_%04x"
```

Comment:
This change came from upstream, where they made it because PCRE 8 outlawed named groups where the name was all digits.

When I was testing (before I knew PCRE named groups weren't allowed) I passed in a named group whose name started with an `f`. `scanf` picked it up and started looking for the Grok group with index `15`, because the format string will take any leading hex digits. This led to unexpected behaviour.

Even if we document that PCRE named groups aren't allowed, this change makes it harder for users to make weird things happen.